src/theory/bv/int_blaster.cpp

```diff
@@ -0,0 +1,1133 @@
+      // original = a[i:j]
+      uint64_t i = bv::utils::getExtractHigh(original);
+      uint64_t j = bv::utils::getExtractLow(original);
+      Assert(i >= j);
+      Node div = d_nm->mkNode(
+          kind::INTS_DIVISION_TOTAL, translated_children[0], pow2(j));
+      returnNode = modpow2(div, i - j + 1);
+      break;
+    }
+    case kind::EQUAL:
+    {
+      returnNode = d_nm->mkNode(kind::EQUAL, translated_children);
+      break;
+    }
+    case kind::BITVECTOR_ULT:
+    {
+      returnNode = d_nm->mkNode(kind::LT, translated_children);
+      break;
+    }
+    case kind::BITVECTOR_ULE:
+    {
+      returnNode = d_nm->mkNode(kind::LEQ, translated_children);
+      break;
+    }
+    case kind::BITVECTOR_UGT:
+    {
+      returnNode = d_nm->mkNode(kind::GT, translated_children);
+      break;
+    }
+    case kind::BITVECTOR_UGE:
+    {
+      returnNode = d_nm->mkNode(kind::GEQ, translated_children);
+      break;
+    }
+    case kind::BITVECTOR_ULTBV:
+    {
+      returnNode = d_nm->mkNode(kind::ITE,
+                                d_nm->mkNode(kind::LT, translated_children),
+                                d_one,
+                                d_zero);
+      break;
+    }
+    case kind::BITVECTOR_SLTBV:
+    {
+      uint64_t bvsize = original[0].getType().getBitVectorSize();
+      returnNode = d_nm->mkNode(
+          kind::ITE,
+          d_nm->mkNode(kind::LT,
+                       unsignedTosigned(translated_children[0], bvsize),
+                       unsignedTosigned(translated_children[1], bvsize)),
+          d_one,
+          d_zero);
+      break;
+    }
+    case kind::ITE:
+    {
+      returnNode = d_nm->mkNode(oldKind, translated_children);
+      break;
+    }
+    case kind::APPLY_UF:
+    {
+      /**
+       * higher order logic allows comparing between functions
+       * The translation does not support this,
+       * as the translated functions may be different outside
+       * of the bounds that were relevant for the original
+       * bit-vectors.
+       */
+      if (childrenTypesChanged(original) && options::ufHo())
+      {
+        throw OptionException("bv-to-int does not support higher order logic ");
+      }
+      // Insert the translated application term to the cache
+      returnNode = d_nm->mkNode(kind::APPLY_UF, translated_children);
+      // Add range constraints if necessary.
+      // If the original range was a BV sort, the original application of
+      // the function Must be within the range determined by the
+      // bitwidth.
+      if (original.getType().isBitVector())
+      {
+        addRangeConstraint(
+            returnNode, original.getType().getBitVectorSize(), lemmas);
+      }
+      break;
+    }
+    case kind::BOUND_VAR_LIST:
+    {
+      returnNode = d_nm->mkNode(oldKind, translated_children);
+      break;
+    }
+    case kind::FORALL:
+    {
+      returnNode = translateQuantifiedFormula(original);
+      break;
+    }
+    case kind::EXISTS:
```

Comment:
Can you turn this into an assert at a higher level?  I believe this will cause a compiler warning as is.